plugins/source/aws/client/client.go

```diff
@@ -373,50 +371,45 @@ func verifyRegions(regions []string) error {
 		return err
 	}
 
+	for _, region := range regions {
+		if region != "*" && !availableRegions[region] {
+			return fmt.Errorf("unsupported region %s", region)
+		}
+	}
+
 	// validate regions values
 	var hasWildcard bool
 	for i, region := range regions {
+		if region == "*" && hasWildcard {
+			return fmt.Errorf("region * can appear only once")
```

Comment:
In Go it's customary to have different error for different places otherwise you can't differentiate where the error came from because this how wrapping works unless we go into the rabbit whole of adding line numbers which is not "Go idiomatic"